WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Payment Methods/ShippingLabelPaymentMethods.swift

```diff
@@ -37,6 +38,7 @@ struct ShippingLabelPaymentMethods: View {
                                           selected: selected)
                             .onTapGesture {
                                 viewModel.didSelectPaymentMethod(withID: method.paymentMethodID)
+                                ServiceLocator.analytics.track(.shippingLabelPurchaseFlow, withProperties: ["state": "payment_method_selected"])
```

Comment:
I have the same question here as with the `packages_selected` event above. Do we want to fire this every time a payment method is selected in the list, or only when the selection is confirmed?

Reply:
Here it makes sense to move it and to fire the event just when the user press `Done` 😉 updated.